Source/mhd/hlld.cpp

```diff
@@ -0,0 +1,437 @@
+#include "Castro.H"
+#include "Castro_F.H"
+
+#include "mhd_util.H"
+
+using namespace amrex;
+
+void
+Castro::hlld(const Box& bx,
+             Array4<Real const> const& qleft,
+             Array4<Real const> const& qright,
+             Array4<Real> const& flx,
+             const int dir) {
+
+  // Riemann solve:
+
+  // Main assumption, the normal velocity/Mag field is constant in the
+  // Riemann fan, and is sM/Bn respectively.  Total Pressure is constant
+  // throughout the Riemann fan, pst!
+
+
+  // `n` here is the normal
+  // `p` are the perpendicular
+
+  int QMAGN, QMAGP1, QMAGP2;
+  int QVELN, QVELP1, QVELP2;
+  int UMN, UMP1, UMP2;
+  int UMAGN, UMAGP1, UMAGP2;
+
+  if (dir == 0) {
+    QMAGN  = QMAGX;
+    QMAGP1 = QMAGY;
+    QMAGP2 = QMAGZ;
+    QVELN  = QU;
+    QVELP1 = QV;
+    QVELP2 = QW;
+    UMN    = UMX;
+    UMP1   = UMY;
+    UMP2   = UMZ;
+    UMAGN  = UMAGX;
+    UMAGP1 = UMAGY;
+    UMAGP2 = UMAGZ;
+
+  } else if (dir == 1) {
+    QMAGN  = QMAGY;
+    QMAGP1 = QMAGZ;
+    QMAGP2 = QMAGX;
+    QVELN  = QV;
+    QVELP1 = QW;
+    QVELP2 = QU;
+    UMN    = UMY;
+    UMP1   = UMZ;
+    UMP2   = UMX;
+    UMAGN  = UMAGY;
+    UMAGP1 = UMAGZ;
+    UMAGP2 = UMAGX;
+
+  } else if (dir == 2) {
+    QMAGN  = QMAGZ;
+    QMAGP1 = QMAGX;
+    QMAGP2 = QMAGY;
+    QVELN  = QW;
+    QVELP1 = QU;
+    QVELP2 = QV;
+    UMN    = UMZ;
+    UMP1   = UMX;
+    UMP2   = UMY;
+    UMAGN  = UMAGZ;
+    UMAGP1 = UMAGX;
+    UMAGP2 = UMAGY;
+  }
+
+  amrex::ParallelFor(bx,
+  [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
+  {
+
+    // this is a loop over interfaces, so, e.g., for dir = 1 (x), we are seeing
```

Comment:
Correct this comment since x is now dir = 0

Reply:
fixed